modules/shortcodes/soundcloud.php

```diff
@@ -109,7 +110,17 @@ function soundcloud_shortcode( $atts, $content = null ) {
 		$options['visual'] = false;
 	}
 
-	// Build our list of Souncloud parameters.
+	if (
+		class_exists( 'Jetpack_AMP_Support' )
+		&& Jetpack_AMP_Support::is_amp_request()
+		&& isset( $options['url'] )
+		&& 'api.soundcloud.com' !== wp_parse_url( $options['url'], PHP_URL_HOST )
```

Comment:
This won't handle shortcodes with `api.soundcloud.com`, like:

```html
[soundcloud url="https://api.soundcloud.com/tracks/156661852" width="400" height="100"]
```

The AMP plugin [has logic](https://github.com/ampproject/amp-wp/blob/develop/includes/embeds/class-amp-soundcloud-embed.php#L144-L149) for converting that to an `<amp-shortcode>`. 

But I didn't include it in this PR, because the AMP plugin still produces a valid `<amp-iframe>` that looks good:
<img width="1101" alt="has-valid-amp-iframe" src="https://user-images.githubusercontent.com/4063887/68832515-c50b9680-0676-11ea-8b8a-a697f90f14a1.png">

I don't think it's worth migrating all of that code for this. The `<amp-soundcloud>` itself also wraps an `<iframe>` very similar to the `<amp-iframe>` above.